container/docker/docker.go

```diff
@@ -161,3 +178,20 @@ func parseDockerVersion(full_version_string string) ([]int, error) {
 	}
 	return version_array, nil
 }
+
+func parseDockerAPIVersion(full_api_version_string string) ([]int, error) {
```

Comment:
To reduce the duplication here, could you create a method:
parseVersion(version_string string, regex re.*Regexp, length int) ([]int, error) {
that could be used by both parseDockerAPIVersion, and ParseDockerVerison?